File: src/main/java/src/SistemaGestionBiblioteca/Biblioteca.java
package SistemaGestionBiblioteca;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Biblioteca {

    private List<Libro> libros;
    private List<Usuario> usuarios;
    private Map<Integer, Integer> prestamos;

    public Biblioteca() {
        this.libros = new ArrayList<>();
        this.usuarios = new ArrayList<>();
        this.prestamos = new HashMap<>();
    }

    public void agregarLibro(Libro libro) {
        libros.add(libro);
        System.out.println("El libro " + libro.getTitulo() + " ha sido agregado.");
    }

    public void registrarUsuario(Usuario usuario) {
        usuarios.add(usuario);
        System.out.println("El usuario " + usuario.getNombre() + " con id " + usuario.getId() + " ha sido registrado.");
    }

    public void prestarLibro(int idLibro, int idUsuario) {
        Libro libro = null;
        Usuario usuario = null;

        for (Libro libroBuscar : libros) {
            if (libroBuscar.getId() == idLibro) {
                libro = libroBuscar;
                break;
            }
        }

        for (Usuario usuarioBuscar : usuarios) {
            if (usuarioBuscar.getId() == idUsuario) {
                usuario = usuarioBuscar;
                break;
            }
        }

        if (libro == null) {
            System.out.println("El libro con id " + idLibro + " no existe.");
        } else if (usuario == null) {
            System.out.println("El usuario con id " + idUsuario + " no está registrado.");
        } else if (libro.isPrestado()){
            System.out.println("El libro " + libro.getTitulo() + " ya está prestado.");
        } else  {
            libro.setPrestado(true);
            prestamos.put(idLibro, idUsuario);
            System.out.println("El libro " + libro.getTitulo() + " ha sido prestado a " + usuario.getNombre() + ".");
        }
    }

    public void devolverLibro(int idLibro, int idUsuario) {
        Libro libro = null;

        for (Libro libroBuscar : libros) {
            if (libroBuscar.getId() == idLibro) {
                libro = libroBuscar;
                break;
            }
        }

        if (libro == null) {
            System.out.println("El libro con id " + idLibro + " no existe.");
        } else if (!libro.isPrestado()){
            System.out.println("El libro " + libro.getTitulo() + " no está prestado.");
        } else  {
            libro.setPrestado(false);
            prestamos.remove(idLibro);
            System.out.println("El libro " + libro.getTitulo() + " ha sido devuelto.");
        }
    }

    public void mostrarLibros() {
        System.out.println("Libros:");
        for (Libro libro : libros) {
            System.out.println(libro);
        }
    }

    public void mostrarUsuarios() {
        System.out.println("Usuarios:");
        for (Usuario usuario : usuarios) {
            System.out.println(usuario);
        }
    }
}

File: src/main/kotlin/SistemaGestionBiblioteca/Biblioteca.kt
package org.example.SistemaGestionBiblioteca

class Biblioteca {
    private val libros: MutableList<Libro> = mutableListOf()
    private val usuarios: MutableList<Usuario> = mutableListOf()
    private val prestamos: MutableMap<Int, Int> = mutableMapOf()

    fun agregarLibro(libro: Libro) {
        libros.add(libro)
        println("El libro " + libro.titulo + " ha sido agregado.")
    }

    fun registrarUsuario(usuario: Usuario) {
        usuarios.add(usuario)
        println("El usuario " + usuario.nombre + " con id " + usuario.id + " ha sido registrado.")
    }

    fun prestarLibro(idLibro: Int, idUsuario: Int) {
        var libro: Libro? = null
        var usuario: Usuario? = null

        for (libroBuscar in libros) {
            if (libroBuscar.id == idLibro) {
                libro = libroBuscar
                break
            }
        }

        for (usuarioBuscar in usuarios) {
            if (usuarioBuscar.id == idUsuario) {
                usuario = usuarioBuscar
                break
            }
        }

        if (libro == null) {
            println("El libro con id $idLibro no existe.")
        } else if (usuario == null) {
            println("El usuario con id $idUsuario no está registrado.")
        } else if (libro.prestado) {
            println("El libro " + libro.titulo + " ya está prestado.")
        } else {
            libro.switchPrestado(true)
            prestamos[idLibro] = idUsuario
            println("El libro " + libro.titulo + " ha sido prestado a " + usuario.nombre + ".")
        }
    }

    fun devolverLibro(idLibro: Int, idUsuario: Int) {
        var libro: Libro? = null

        for (libroBuscar in libros) {
            if (libroBuscar.id == idLibro) {
                libro = libroBuscar
                break
            }
        }

        if (libro == null) {
            println("El libro con id $idLibro no existe.")
        } else if (!libro.prestado) {
            println("El libro " + libro.titulo + " no está prestado.")
        } else {
            libro.switchPrestado(false)
            prestamos.remove(idLibro)
            println("El libro " + libro.titulo + " ha sido devuelto.")
        }
    }

    fun mostrarLibros() {
        println("Libros:")
        for (libro in libros) {
            println(libro)
        }
    }

    fun mostrarUsuarios() {
        println("Usuarios:")
        for (usuario in usuarios) {
            println(usuario)
        }
    }
}